src/main/java/com/github/manolo8/darkbot/utils/login/LoginUtils.java

```diff
@@ -118,17 +117,37 @@ public static void usernameLogin(LoginData loginData) {
     }
 
     public static void usernameLogin(LoginData loginData, String domain) {
-        String loginUrl = Http.create("https://" + domain + ".darkorbit.com/")
-                .consumeInputStream(LoginUtils::getLoginUrl);
+        String frontPage = "";
+        String gResponse = "";
+        try {
+            frontPage = IOUtils.read(Http.create("https://" + domain + ".darkorbit.com/").getInputStream());
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
```

Comment:
```suggestion
        String frontPage = IOUtils.read(Http.create("https://" + domain + ".darkorbit.com/").getInputStream());
```

If this is the exception handling you're gonna do (none) better not do it at all, just bubble the exception up instead.

Make this method throw IOException, let the higher-up usernameLogin method handle the exception (by trying a diff domain).

Also make this method private if possible, it's an implentation detail